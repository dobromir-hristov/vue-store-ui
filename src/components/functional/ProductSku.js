import { mapGetters } from 'vuex'

export const ProductSku = {
  computed: {
    ...mapGetters('product/productForm', ['sku'])
  }
}
