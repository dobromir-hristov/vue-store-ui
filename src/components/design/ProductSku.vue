<template>
  <div class="ProductSku text-grey text-sm">
    SKU: <span class="uppercase">{{ sku }}</span>
  </div>
</template>

<script>
import { ProductSku } from '@/components/functional/ProductSku'

/**
 * @module ProductSku
 */
export default {
  name: 'ProductSku',
  mixins: [ProductSku]
}
</script>
