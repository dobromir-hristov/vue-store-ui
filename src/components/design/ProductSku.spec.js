import ProductSku from './ProductSku'

/**
 * @param overrides
 * @returns {{config, wrapper: Wrapper<V> | Wrapper<Vue> | *}}
 */
const createWrapper = (overrides) => {
  let store = {
    'product/productForm': {
      getters: {
        sku: () => 'what-ev-er'
      }
    }
  }
  let config = _merge(createComponentMocks({ store }), overrides)
  const wrapper = mount(ProductSku, config)
  return {
    config,
    wrapper
  }
}

describe('ProductSku', () => {
  it('renders the ProductSku', () => {
    const { wrapper } = createWrapper()
    expect(wrapper.text()).toContain('what-ev-er')
    expect(wrapper.html()).toMatchSnapshot()
  })
})
