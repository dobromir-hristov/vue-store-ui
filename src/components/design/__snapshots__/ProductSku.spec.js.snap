// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductSku renders the ProductSku 1`] = `
<div class="ProductSku text-grey text-sm">
  SKU: <span class="uppercase">what-ev-er</span></div>
`;
